### PR TITLE
Add new docs structure

### PR DESCRIPTION
**Pull request by iHiD**
Over the last couple of weeks, Erik and I have been working in the background on restructuring the v3 repo.

It's going to serve two functions:
1) Be a central point of contribution for general contributors helping with v3. The `README.md`, `docs/*.md` and GitHub issues will all be primarily for this purpose.
2) Be a hub for maintainers working on their tracks, with `/languages` and `/reference` containing the existing maintainer-facing reference.

Pretty much all of this existing repo is going to get moved into `/languages` and `reference` as part of this process, and I will follow up with a PR later in the week (probably Tuesday) for that, with some more details on that part. However, as that is going to be a giant and unwieldy PR (ever file gets renamed or moved basically), I thought I would add a separate PR first to add the new README and some docs.

The idea here is:
- Any contributor will be points to the main `README.md`, which should act as an introduction to v3 and how they can help. The other files in `docs/.md` are supporting information that will be linked from GitHub issues and the README. 
- The `docs/maintainers/README.md` acts as a further introduction to maintainers, with the `docs/maintainers/*.md` acting as further resources for maintainers getting started.

These docs will evolve over the coming days and weeks (hopefully as maintainers get stuck in and clarify things that are missing/unclear), but I'd like reviews on whether there is anything obviously unclear or missing with this as a starting point. Thanks!

**Note: I recommend reading the [rendered README.md](https://github.com/exercism/v3/blob/5adc4be9a3d91977cee3408bdb5e4837c61f554e/README.md) as a starting point.**

**Note 2: There will be broken links, because this PR is a subset of the whole PR. If a link is broken to a file that this PR touches, please tell me. Otherwise, we'll get the links all green before sending over the main PR. We won't be merging this directly into the repo, but backport any changes into our working repo before the main PR**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,109 +1,31 @@
...
-- If someone learnt Ruby, and someone learnt Haskell, what are the things that the two people learnt that are different?
-- If a Ruby programmer learnt Haskell, what new concepts would they have to learn, what knowledge would they have to disregard, and what syntax would they have to remap?
+**If you are a Track Maintainer, please [start here](./docs/maintainers/README.md).**
 
-### What do we mean by "fluency?"
+## More information
 
-By "Fluency", we mean: Do you **get** that language? Can you reason in that language? Do you write that language like a "native" programmer writes in it? Fluency is the ability to express oneself articulately in a particular language.
+- [The features of v3](./docs/features-of-v3.md): A deeper dive into the changes we're making to v3.
+- [Rationale for v3](./docs/rationale-for-v3.md). An explanation about the rational behind some of our our design decisions for v3.
```
Comment (ErikSchierboom) [inline README.md]:
```suggestion
- [Rationale for v3](./docs/rationale-for-v3.md). An explanation about the rationale behind some of our our design decisions for v3.
```

File: README.md [hunk 2]
```diff
@@ -1,109 +1,31 @@
...
-It can also be helpful to group related concepts. For example, the Classes, Polymorphism and Inheritance concepts are all object-oriented concepts, which can then be grouped as "Object-oriented concepts".
-
-Filling out the concepts will be iterative, it is hard to get this right immediately. While creating concepts exercises, you'll probably find missing concepts or perhaps want to split out concepts.
-
-### Writing your first concept exercise
-
-Once a list of concepts has been compiled, these then need to be translated into Concept Exercises. Writing your first concept exercise begins with choosing a concept to write an exercise for. Good first concept candidates are concepts that involve working with basic types, like strings and numbers.
-
-Having chosen a concept, look for the concept's description in this repository. This file should have references to all tracks that have implemented the exercise. You can use these existing implementations as a starting point for your first concept exercise, which you can then tailor to your specific language.
+- [What are concept exercise are and how they are structured?](./docs/concept-exercises.md)
```
Comment (ErikSchierboom) [inline README.md]:
```suggestion
- [What are concept exercise and how they are structured?](./docs/concept-exercises.md)
```

File: README.md [hunk 3]
```diff
@@ -1,109 +1,31 @@
...
+- [How is this repository structured?](./docs/repository-structure.md)
```
Comment (ErikSchierboom) [inline README.md]: The `repository-structure.md` file is in the `docs/maintainers` directory. Looking at the sidebar definitions, I'm starting to think it should be in `docs`, and that this link is actually correct, but the file's location isn't.

File: docs/_sidebar.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+- [Introduction](/)
+- [Docs](README.md)
+  - [The features of v3](features-of-v3.md): A in-depth look at the changes coming to Exercism in v3
+  - [Rationale for v3](rationale-for-v3.md): An article exploring the rational behind our design decisions with v3.
```
Comment (ErikSchierboom) [inline docs/_sidebar.md]:
```suggestion
  - [Rationale for v3](rationale-for-v3.md): An article exploring the rationale behind our design decisions with v3.
```

File: docs/_sidebar.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  - [What are concept exercise are and how they are structured?](concept-exercises.md)
```
Comment (ErikSchierboom) [inline docs/_sidebar.md]:
```suggestion
  - [What are concept exercise and how they are structured?](concept-exercises.md)
```

File: docs/concept-exercises.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+## How are Concept Exercises designed and structured?
+
+Concept Exercises must have the following characteristics:
+
+- Each one has a clear learning goal.
+- They are language-specific, not generic.
+- Stubs/boilerplate are used to avoid the student having to learn/write unnecessary code on exercises.
+
+Concept Exercises are **not** mentored. When a user submits a submission that gets the tests passing for a Concept Exercise, we check for an Analyzer or Representer to give feedback. If none is found, then the solution is approved. This shifts the burden of teaching to the exercise, which must provide a clear pathway to learning the concept that needs learning.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]: Two times learning shortly after each other. Maybe something like?

```suggestion
Concept Exercises are **not** mentored. When a user submits a submission that gets the tests passing for a Concept Exercise, we check for an Analyzer or Representer to give feedback. If none is found, then the solution is approved. This shifts the burden of teaching to the exercise, which must provide a clear pathway to learning the concept that is being taught.
```

File: docs/concept-exercises.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+Exercises are unlocked based on concepts taught and learnt. Each Concept Exercise must teach one or more concepts. It may also have prerequisites on Concepts, which means it will not be unlocked until Concept Exercises teaching those prerequisite concepts have been completed.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]:
```suggestion
Exercises are unlocked based on concepts taught and learnt. Each Concept Exercise must teach one or more concepts. It may also have prerequisites on concepts, which means it will not be unlocked until Concept Exercises teaching those prerequisite concepts have been completed.
```

The lowercase form is used later in the sentence too.

File: docs/concept-exercises.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+Concept Exercises do not share a common base like Practice Exercises do in the `problem-specifications` repository. Instead they "share" concepts that they are teaching with other languages. This repository aims to list all of those concepts and provide information about the concept that maintainers can use as the basis for their own languages. Each concept should also link to the implementations in different languages. Maintainers are free to copy and paste from each others repositories, and then edit to make things specific to their tracks, but such copy-and-pastes should be considered hard-forks.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]: Should we expand on the fact why we are sharing concepts with other languages?

Comment (ErikSchierboom) [inline docs/concept-exercises.md]:
```suggestion
Concept Exercises do not share a common base like Practice Exercises do in the `problem-specifications` repository. Instead they "share" concepts that they are teaching with other languages. This repository aims to list all of those concepts and provide information about the concepts that maintainers can use as the basis for their own languages. Each concept should also link to the implementations in different languages. Maintainers are free to copy and paste from each others repositories, and then edit to make things specific to their tracks, but such copy-and-pastes should be considered hard-forks.
```

File: docs/concept-exercises.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+For example, we might define a concept of "Classes" and provide a short introduction that explains what a class is, how it fits with objects, state, etc. We might include a link to a good article introducing OOP and classes. Individual tracks implementing an exercise on Classes can then include this introductory text, making any additions that explain their language-specific semantics and syntax.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]: Maybe make it explicit that we allow people to modify the shared text when applied to their language?

File: docs/maintainers/README.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
+# Maintainers - How to get started
+
+Firstly, thank you for your willingness to help with Exercism. We believe that v3 will be a huge step forward in our aim to offer a comprehesive free learning website to anyone wanting to develop their programming skills, and we're really excited to see it play out.
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]:
```suggestion
Firstly, thank you for your willingness to help with Exercism. We believe that v3 will be a huge step forward in our aim to offer a comprehensive, free learning website to anyone wanting to develop their programming skills, and we're really excited to see it play out.
```

File: docs/maintainers/README.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+**Before doing anything else, please create an issue in this repository asking for your track to be bootstrapped. You can then spend some time learning about the v3 process while we set things up for you.**
+
+## Concept Exercises
+
+The key things to understand about v3 is that we are making a strong distinction between Concept Exercises and Practice Exercises. All existing exercises will become Practice Exericses - exercises where students can practice the ideas they are learning, and discuss them with a mentor. The new Concept Exercises will be written from scratch, each with the aim of teaching one or more programming concepts that are necessary to become fluent in a given language.
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]:
```suggestion
The key thing to understand about v3 is that we are making a strong distinction between Concept Exercises and Practice Exercises. All existing exercises will become Practice Exericses - exercises where students can practice the ideas they are learning, and discuss them with a mentor. The new Concept Exercises will be written from scratch, each with the aim of teaching one or more programming concepts that are necessary to become fluent in a given language.
```

Maybe instead of `can practice the ideas they are learning` do `can practice the concepts they are learning`?

File: docs/maintainers/README.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+The existing exercises are temporarily removed from the `config.json` file. They will return as practice exercises once the concept exercises have been added.
+
+TODO: Discuss more about what needs to be done
+
+
+## Tooling
+
+As well as building out the Concept Exercises, your track will also need a Test Runner and a Representer.
+
+The Test Runner is a piece of software that runs the tests. This enables us to have in-browser coding. Test Runners are surprisingly tricky to get right, with lots of weird edge cases and oddities, so we recommend starting on them quite early. You can find more in the [automated testing repository](https://github.com/exercism/automated-tests).
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]:
```suggestion
The Test Runner is a piece of software that runs the tests. This enables us to have in-browser coding. Test Runners are surprisingly tricky to get right, with lots of weird edge cases and oddities, so we recommend starting on them quite early. You can learn more about Test Runners in the [automated testing repository](https://github.com/exercism/automated-tests).
```

File: docs/maintainers/README.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+Representers are pieces of software that take a solution and provide a normalised representation of it. We can then attach feedback to those representations, and automatically provide the same feedback for future solutions that normalise to the same representation. You may also like to create an Analyzer, which automatically provides feedback based on heursitics provided by AST matching. You can learn more about Representers and Analysers in our [automated analysis repository](https://github.com/exercism/automated-analysis).
+
+## Getting help
+
+Each track has a #maintaining-x channel on [Slack](https://exercism-team.slack.com). We recommend using that to chat amongst yourselves.
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]: I'm wondering a little bit if we should perhaps also encourage people to post comments on the GH issues, as then the discussion becomes public and is automatically archived.

File: docs/maintainers/determining-concepts.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
+# Maintainers - Choosing and defining Concepts
+
+It is worth reading the start of the [Concept Exercises](concept-exercises.md) file to familiarise yourself with Concept Exercises and why writing out concepts is important.
+
+To know which Concept Exercises should exist, a list of the language's concepts should first be compiled. There are various resources to use when compiling this list: books, (official) documentation, your own experiences and the [concepts listed in this repository](../reference/concepts/README.md).
+
+As many languages use the same concepts (but possibly implemented differently), it can greatly help to look at what other languages have already done in this regard. See the [C# concepts](../languages/csharp/reference/concepts/README.md) for an example.
```
Comment (ErikSchierboom) [inline docs/maintainers/determining-concepts.md]:
```suggestion
As many languages use the same concepts (but possibly implemented differently), it can greatly help to look at what other languages have already done in this regard. See the [C# concepts](../languages/csharp/reference/README.md) for an example.
```

File: docs/maintainers/README.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+  "concept": [],
+  "practice": []
+}
+```
+
+4. Added concepts for all Practice Exercises
+
+The existing exercises are temporarily removed from the `config.json` file. They will return as practice exercises once the concept exercises have been added.
+
+TODO: Discuss more about what needs to be done
```
Comment (ErikSchierboom) [inline docs/maintainers/README.md]: @iHiD Do we need to write this bit already? I know what I want the new format to look like, so I could do that.

File: docs/concept-exercises.md (hunk 8 above)
Comment (iHiD) [reply docs/concept-exercises.md]: @ErikSchierboom I can't see the difference.

File: docs/concept-exercises.md (hunk 9 above)
Comment (iHiD) [reply docs/concept-exercises.md]:
```suggestion
For example, we might define a concept of "Classes" and provide a short introduction that explains what a class is, how it fits with objects, state, etc. We might include a link to a good article introducing OOP and classes. Individual tracks implementing an exercise on Classes can then include this introductory text, making any changes or additions that explain their language-specific semantics and syntax.
```

File: docs/concept-exercises.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+Exercises are unlocked based on concepts taught and learnt. Each Concept Exercise must teach one or more concepts. It may also have prerequisites on Concepts, which means it will not be unlocked until Concept Exercises teaching those prerequisite concepts have been completed.
+
+Concept Exercises do not share a common base like Practice Exercises do in the `problem-specifications` repository. Instead they "share" concepts that they are teaching with other languages. This repository aims to list all of those concepts and provide information about the concept that maintainers can use as the basis for their own languages. Each concept should also link to the implementations in different languages. Maintainers are free to copy and paste from each others repositories, and then edit to make things specific to their tracks, but such copy-and-pastes should be considered hard-forks.
```
Comment (iHiD) [inline docs/concept-exercises.md]:
```suggestion
Concept Exercises do not share a common base like Practice Exercises do in the `problem-specifications` repository. Instead they "share" Concepts that they are teaching with other languages. This repository aims to list all of those Concepts and provide information about the Concept that maintainers can use as the basis for their own languages. Each Concept should also link to the implementations in different languages. Maintainers are free to copy and paste from each others repositories, and then edit to make things specific to their tracks, but such copy-and-pastes should be considered hard-forks.
```

**Comment by iHiD** [comment]
Cherry-picked into #28.
